sqlglot/dialects/singlestore.py

```diff
@@ -1166,3 +1246,98 @@ class Generator(MySQL.Generator):
             "zerofill",
             "zone",
         }
+
+        def jsonpathkey_sql(self, expression: exp.JSONPathKey) -> str:
```

Comment:
Why do we need to override the generators for these `JSONPathPart` subclasses?

Reply:
In SingleStore, the JSON path is provided as a comma-separated list of keys/indexes.
For example, instead of `$.b[2]`, SingleStore expects `'b', 2`.
To support this, I overrode `jsonpathkey_sql` and `jsonpathsubscript_sql`.

For all other JSON path parts that SingleStore does not support, I added an explicit "not supported" message, for example:
```
self.unsupported("JSONPathRoot is not supported in SingleStore")
```

Reply:
> In SingleStore, the JSON path is provided as a comma-separated list of keys/indexes.

Can you check whether `json_extract_segments` does the trick? We do something similar for Postgres: https://github.com/tobymao/sqlglot/blob/main/sqlglot/dialects/postgres.py#L188-L196 (see [test](https://github.com/tobymao/sqlglot/blob/902a0cdfe46f693aa55612d45a2de2def21f0b8c/tests/dialects/test_postgres.py#L464)).

Reply:
`json_extract_segments` leaves redundant `$`.
```
Expected :SELECT JSON_EXTRACT_JSON(a, 'b') FROM t
Actual   :SELECT JSON_EXTRACT_JSON(a, '$', ''b'') FROM t
```

Reply:
Perhaps we can switch of the `JSONPathRoot` to avoid this? E.g., for this reason Postgres has the following in its `Generator.TRANSFORMS` mapping:

```python
exp.JSONPathRoot: lambda *_: "",
```

Not sure if the `$...` syntax is used elsewhere for Singlestore.